packages/components/src/internal/query/api.ts

```diff
@@ -934,31 +921,106 @@ export function deleteRows(options: DeleteRowsOptions): Promise<DeleteRowsRespon
             queryName: schemaQuery.queryName,
             success: response => {
                 resolve(
-                    Object.assign(
-                        {},
-                        {
-                            schemaQuery: options.schemaQuery,
-                            rows: response.rows,
-                            transactionAuditId: response.transactionAuditId,
-                        }
-                    )
+                    new QueryCommandResponse({
+                        schemaQuery: options.schemaQuery,
+                        rows: response.rows,
+                        transactionAuditId: response.transactionAuditId,
+                    })
                 );
             },
             failure: error => {
-                reject(
-                    Object.assign(
-                        {},
-                        {
-                            schemaQuery: options.schemaQuery,
-                        },
-                        error
-                    )
-                );
+                reject({
+                    schemaQuery: options.schemaQuery,
+                    error,
+                });
             },
         });
     });
 }
 
+export function saveRows(options: Query.SaveRowsOptions): Promise<Query.SaveRowsResponse> {
+    return new Promise((resolve, reject) => {
+        Query.saveRows({
+            apiVersion: 13.2,
+            ...options,
+            success: response => {
+                resolve(response);
+            },
+            failure: error => {
+                reject(error);
+            },
+        });
+    });
+}
+
+export function deleteRowsByContainer(
+    options: DeleteRowsOptions,
+    containerField: string = 'ContainerPath'
+): Promise<QueryCommandResponse> {
+    const commands = [];
+
+    const allRows = options.rows;
+    if (
+        !allRows ||
+        Object.keys(allRows[0])
+            .map(key => key.toLowerCase())
+            .indexOf(containerField.toLowerCase()) === -1
+    )
+        return deleteRows(options);
+
+    const containerRows = {};
+    allRows.forEach(row => {
+        const container = caseInsensitive(row, containerField);
+        if (!containerRows[container]) containerRows[container] = [];
+        containerRows[container].push(row);
+    });
+
+    if (Object.keys(containerRows).length <= 1) {
```

Comment:
same comment here as above. isn't this an error case if containerRows is empty?

Reply:
containerRows shouldn't be empty, otherwise the deleteRows call above would be used. the <= 1 can be changed to == 1.